packages/transformers/jsonld/src/JSONLDTransformer.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import {Transformer} from '@parcel/plugin';
+import json5 from 'json5';
+
+// A list of all attributes in a schema that may produce a dependency
+// Based on https://schema.org/ImageObject
+// Section "Instances of ImageObject may appear as values for the following properties"
+const SCHEMA_ATTRS = [
+  'logo',
+  'photo',
+  'image',
+  'thumbnail',
+  'screenshot',
+  'primaryImageOfPage',
+  'embedUrl',
+  'thumbnailUrl',
+  'video',
+  'contentUrl',
+  'url'
+];
+
+export default new Transformer({
+  async transform({ asset, config, logger, options, resolve }) {
+    
+    let rawCode = await asset.getCode();
+    // allowing any recieved jsonld to be in json5 format
+    let jsonCode = json5.parse(rawCode);
+    
+    let parser = new JSONLDParser(asset);
+    parser.parse(jsonCode);
+
+    // json should be injected back into the html page
+    asset.type = 'html';
```

Comment:
as @mischnic mentioned [here](https://github.com/parcel-bundler/parcel/pull/4397#discussion_r399711888) the `asset.type` should be set to json. However, when it is set to that, the json does not get outputted to the index.html file in the proper format, which is what it was doing in v1. How should I go about getting this fixed. Should this be handled in a different PR / Issue?

Reply:
@mischnic & @DeMoorJasper what should I do here? Should I change it to 'json' for this PR or keep it as is?

Reply:
another thing that isn't clear to me -- is parcel 2 supposed to output the jsonld into its own json file for the bundle dist or should it output back into the bundled index.html file like parcel v1 does?

Reply:
use `asset.type = 'json';`?

Reply:
> use asset.type = 'json';?

Then the json transformer will output a js asset... 
And implementing the change to run that only for js assets is non-trivial.

> is parcel 2 supposed to output the jsonld into its own json file for the bundle dist or should it output back into the bundled index.html file like parcel v1 does?

If the jsonld was inlined in a html file, it will be inserted back again.
`<script src="..." type="application/ld+json"></script>` would create a separate bundle.

Reply:
@DeMoorJasper & @mischnic but when I set it to `asset.type = "json"` it does NOT insert the json back into the script tag. Either it inserts a `<div></div>` inside the script tag or it inserts it into the script tag as if it were a js asset with `module.exports=...` syntax, which is again not what parcel v1 does.

Reply:
> If the jsonld was inlined in a html file, it will be inserted back again.
> `<script src="..." type="application/ld+json"></script>` would create a separate bundle.
thats good to know and makes perfect sense

so again, in the unit test case, the jsonld IS inlined html and parcel v2 is NOT inserting the transformed jsonld object back into the script tag. See my previous comment.